core/src/androidMain/kotlin/Peripheral.kt

```diff
@@ -254,12 +258,24 @@ public class AndroidPeripheral internal constructor(
 
     private suspend fun discoverServices() {
         logger.verbose { message = "discoverServices" }
-        connection.execute<OnServicesDiscovered> {
-            discoverServices()
-        }
-        _discoveredServices = withContext(connection.dispatcher) {
-            connection.bluetoothGatt.services.map(::DiscoveredService)
+
+        repeat(DISCOVER_SERVICES_RETRIES) { attempt ->
+            connection.execute<OnServicesDiscovered> {
+                discoverServices()
```

Comment:
Is it ok to retry in quick succession with no delay?

Reply:
I _think_ so. But good point. Android BLE always seems pretty delicate in that regard. I'll add a short delay. 👍

Reply:
Thought about this some more, I don't _think_ there is any need to add an artificial delay.

Partly because the `connection.execute` is having to do thread switching so adds a very slight delay anyways, but also I don't really want to add a delay prematurely if it isn't necessary (plus, I don't even know what delay would be appropriate). I figure we'll try this out, and if we still see issues on our internal libraries, then I'll check logs to see if it was super fast failures — one after the other.

It is also only 5 attempts, so it isn't like it is going to flood the system with requests. 🤷